src/lib/routers/history.ts

```diff
@@ -67,19 +112,24 @@ class BrowserHistory {
   }
 
   /**
-   * This method pushes a search state into the URL.
-   * @param {object} routeState a syncable UI state
-   * @return {undefined}
+   * Reads the URL and returns a syncable UI search state.
    */
-  write(routeState) {
+  private read(): UiState {
+    return this.parseURL({ qsModule: qs, location: window.location });
+  }
+
+  /**
+   * Pushes a search state into the URL.
+   */
+  private write(routeState: UiState): void {
     const url = this.createURL(routeState);
     const title = this.windowTitle && this.windowTitle(routeState);
 
     if (this.writeTimer) {
       window.clearTimeout(this.writeTimer);
     }
 
-    this.writeTimer = setTimeout(() => {
+    this.writeTimer = window.setTimeout(() => {
```

Comment:
this code now will no longer run on node, is there a reason you added window.?

Reply:
TypeScript types this as `NodeJS.Timer` otherwise, which is not assignable to a number. See can [read more on this thread](https://github.com/Microsoft/TypeScript/issues/842).

Otherwise we can keep the previous version and type `writeTimer` as any.

Reply:
We don't actually care what the type of the number is, I'd put it as any

Reply:
Note that we rely on `window` everywhere else in this file, so we might want to change that later on. Besides, this is a browser history.